src/PasswordHasher/Md5Hasher.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+include_once("src/PasswordHasher.php");
+
+class md5Hasher implements PasswordHasher {
+
+    /**
+     * 
+     * @param string $password
+     * @param User $user
+     */
+    public function setPassword($password, $user) {
+
+        $uniqueSalt = $this->getUniqueSalt();
+        $hashedPassword = $uniqueSalt . $password;
+        $n = 0;
+        $start = microtime(true);
+
+        while (microtime(true) - $start < 1) {
+            $hashedPassword = md5($hashedPassword);
+            $n++;
+        }
+
+        $user->setSalt($uniqueSalt)
+                ->setHashTimes($n)
+                ->setHashedPassword();
+        return;
+    }
+
+    /**
+     * 
+     * @param string $password
+     * @param User $user
+     * @return boolean
+     */
+    public function passwordsEqual($password, $user) {
+
+        $uniqueSalt = $user->getSalt();
+        $hashedPassword = $uniqueSalt . $password;
+        $n = $user->getHashTimes();
+
+        for ($i = 0; $i < $n; $i++) {
```

Comment:
Kokia prasmė hash times priskirti atskiram kintamajam `$n`? Prasmė būna jeigu yra labai ilga išraiška ir jos rezultatą priskyrus trumpavardžiui kintamąjam paskui viskas sutelpa į trumpesnes eilutes (ilga eilutė - bent 80-100 simbolių).
Kitas variantas su prasme - jeigu išraiška kartojasi daug kartų. Tada vėlgi gerokai sutrumpėja kodas. Šiuo atveju dėl eilučių ilgio problemų nėra, o kodas bus net trumpesnis ir greitesnis jeigu nenaudosi `$n`. Tas pats su `$uniqueSalt`

Reply:
Dėl $uniqueSalt viskas tvarkoje, pilnai su tavim sutinku, bet dėl kintamojo $n norėčiau paprieštaraut, kadangi jis naudojamas cikle ir jei vietoj $n bus objekto funkcija ir programa kiekvieną kartą sukant ciklą kreipsis į objektą norėdama sužinoti ar dar jai sukti sekantį ciklą, to pasekoje, programa turėtų ilgiau dirbti. Aišku, šiuo atveju , gal labai programos darbas neprailgs, bet man atrodo geriau turėti kuo greitesnę programą, nei neturėti papildomos eilutės, be to kažkada su tavim susitikę kalbėjom, kai funkcija įstatoma vietoj $n kintamojo. Kaip tu manai šiuo klausimu ?

Reply:
You made your point. Labai džiaugiuosi, kad sugebi pagrįsti savo sprendimą ir šiuo atveju esi visiškai teisus